src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ArgumentObject.cs

```diff
@@ -2,15 +2,22 @@
 
 namespace Microsoft.CSharp.RuntimeBinder
 {
-    // This class is used to keep the tuple of runtime object values and 
+    // This struct is used to keep the tuple of runtime object values and 
     // the type that we want to use for the argument. This is different than the runtime
     // value's type because unless the static time type was dynamic, we want to use the
     // static time type. Also, we may have null values, in which case we would not be 
     // able to get the type.
-    internal sealed class ArgumentObject
+    internal struct ArgumentObject
     {
-        internal Type Type;
-        internal object Value;
-        internal CSharpArgumentInfo Info;
+        internal readonly Type Type;
+        internal readonly object Value;
+        internal readonly CSharpArgumentInfo Info;
+
+        public ArgumentObject(object value, CSharpArgumentInfo info, Type type)
+        {
+            Value = value;
+            Info = info;
+            Type = type;
```

Comment:
Nit: doesn't matter to me the order of Type, Value, and Info, but it'd be nice if the ordering amongst the fields, the parameters, and the assignments was consistent